hack/quickstart-setup.sh

```diff
@@ -0,0 +1,322 @@
+    echo "Waiting for external-dns deployments to be ready"
+    kubectl -n external-dns wait --timeout=300s --for=condition=Available deployments --all
+    echo "Waiting for cert-manager deployments to be ready"
+    kubectl -n cert-manager wait --timeout=300s --for=condition=Available deployments --all
+}
+
+deployQuickStartSpoke() {
+   clusterName=${1}
+    kubectl config use-context kind-${clusterName}
+    echo "Initialize quickstart setup on ${clusterName}"
+    kustomize build ${QUICK_START_SPOKE_KUSTOMIZATION} --enable-helm --helm-command helm | kubectl apply -f -
+    echo "Waiting for ingress-nginx deployments to be ready"
+    kubectl -n ingress-nginx wait --timeout=600s --for=condition=Available deployments --all
+    echo "Waiting for metallb-system deployments to be ready"
+    kubectl -n metallb-system wait --for=condition=ready pod --selector=app=metallb --timeout=300s
+}
+
+configureControlCluster() {
+
+    clusterName=${1}
+    # Ensure the current context points to the control plane cluster
+    kubectl config use-context kind-${clusterName}
+    kubectl label managedcluster kind-mgc-control-plane ingress-cluster=true
+    cat <<EOF | kubectl apply -f -
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: GatewayClass
+metadata:
+  name: kuadrant-multi-cluster-gateway-instance-per-cluster
+spec:
+  controllerName: "kuadrant.io/mgc-gw-controller"
+EOF
+}
+
+
+# Prompt user for any required env vars that have not been set
+if [[ -z "${MGC_AWS_ACCESS_KEY_ID}" ]]; then
+  echo "Enter your AWS secret access key id:"
+  read MGC_AWS_ACCESS_KEY_ID </dev/tty
+  echo "export MGC_AWS_ACCESS_KEY_ID for future executions of the script to skip this step"
+fi
+
+if [[ -z "${MGC_AWS_SECRET_ACCESS_KEY}" ]]; then
+  echo "Enter your AWS secret access key:"
+  read MGC_AWS_SECRET_ACCESS_KEY </dev/tty
+  echo "export MGC_AWS_SECRET_ACCESS_KEY for future executions of the script to skip this step"
+fi
+
+if [[ -z "${MGC_AWS_SECRET_ACCESS_KEY}" ]]; then
+  echo "Enter an AWS region (e.g. eu-west-1):"
+  read MGC_AWS_SECRET_ACCESS_KEY </dev/tty
+  echo "export MGC_AWS_SECRET_ACCESS_KEY for future executions of the script to skip this step"
+fi
+
+if [[ -z "${MGC_AWS_DNS_PUBLIC_ZONE_ID}" ]]; then
+  echo "Enter the Public Zone ID of your Route53 zone:"
+  read MGC_AWS_DNS_PUBLIC_ZONE_ID </dev/tty
+  echo "export MGC_AWS_DNS_PUBLIC_ZONE_ID for future executions of the script to skip this step"
+fi
+
+if [[ -z "${MGC_ZONE_ROOT_DOMAIN}" ]]; then
+  echo "Enter the root domain of your Route53 hosted zone (e.g. www.example.com):"
+  read MGC_ZONE_ROOT_DOMAIN </dev/tty
+  echo "export MGC_ZONE_ROOT_DOMAIN for future executions of the script to skip this step"
+fi
+
+# Default config
+if [[ -z "${LOG_LEVEL}" ]]; then
+  LOG_LEVEL=1
+fi
+if [[ -z "${OCM_SINGLE}" ]]; then
+  OCM_SINGLE=true
+fi
+if [[ -z "${MGC_WORKLOAD_CLUSTERS_COUNT}" ]]; then
+  MGC_WORKLOAD_CLUSTERS_COUNT=1
+fi
```

Comment:
For this quickstart, can we be opinionated about these settings rather than exposing them?

Reply:
@mikenairn @maleck13 wdyt?

Reply:
yes we can